### PR TITLE
Add a createReaderFromFile method to the ReaderEntityFactory

### PR DESCRIPTION
**Pull request by madflow**
*  create a reader by file type #569 

* The copy and paste in the tests is a "shame on me". Maybe there is a clever way of avoiding the code duplication. I think this should be subject to a future refactoring.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @madflow has signed the CLA. Thanks for the pull request!

File: docs/_pages/getting-started.md [hunk 1]
```diff
@@ -68,6 +68,17 @@ $reader->close();
 
 If there are multiple sheets in the file, the reader will read all of them sequentially.
 
+---
+
+I addition to passing a reader type to ```ReaderEntityFactory::createReader``` - it is also possible to provide a path to a file and create the reader.
```
Comment (adrilo) [inline docs/_pages/getting-started.md]: I**n** addition. Also replace `-` by a comma

File: src/Spout/Reader/Common/Creator/ReaderFactory.php [hunk 2]
```diff
@@ -46,6 +57,34 @@ public static function create($readerType)
...
+                sprintf('Could not open "%s" for reading! File does not exist.', $path)
+            );
+        }
+
+        $ext = \pathinfo($path, PATHINFO_EXTENSION);
+        $ext = \strtolower($ext);
+        $readerType = self::$extensionReaderMap[$ext] ?? null;
+        if ($readerType === null) {
+            throw new UnsupportedTypeException(
+                sprintf('No readers support the file extension "%s".', $ext)
```
Comment (adrilo) [inline src/Spout/Reader/Common/Creator/ReaderFactory.php]: for consistency: `support` => `supporting`

File: tests/Spout/Reader/Common/Creator/ReaderEntityFactoryTest.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+use PHPUnit\Framework\TestCase;
+
+class ReaderEntityFactoryTest extends TestCase
+{
+    use TestUsingResource;
+
+    /**
+     * @return void
+     */
+    public function testCreateFromFileCSV()
```
Comment (adrilo) [inline tests/Spout/Reader/Common/Creator/ReaderEntityFactoryTest.php]: nitpicky: I would group the valid and invalid cases together

File: tests/Spout/Reader/Common/Creator/ReaderFactoryTest.php [hunk 4]
```diff
@@ -19,4 +23,64 @@ public function testCreateReaderShouldThrowWithUnsupportedType()
 
         ReaderFactory::create('unsupportedType');
     }
+
+    /**
+     * @return void
+     */
+    public function testCreateFromFileCSV()
```
Comment (adrilo) [inline tests/Spout/Reader/Common/Creator/ReaderFactoryTest.php]: same here

**Comment by adrilo** [comment]
Copy and paste in the tests => it's ok for now.
Ideally, we should do the same thing for writers :)

**Comment by madflow** [comment]
I amended the PR in regards to the change requests and rebased. The travis build passed in my fork - here it show some esoteric message.

**Comment by adrilo** [comment]
I restarted the job and it passed :)

**Pull request merged**